api/src/main/java/org/openmrs/module/initializer/api/BaseAttributeLineProcessor.java

```diff
@@ -28,32 +28,64 @@ public T fill(T instance, CsvLine line) throws IllegalArgumentException {
 		
 		Customizable<A> attributable = (Customizable<A>) instance;
 		
-		Consumer<? super SimpleEntry<String, String>> processAttributeData = attData -> {
-			
-			AT attType = getAttributeType(attData.getKey());
-			if (attType == null) {
-				throw new IllegalArgumentException("An attribute value is specified ('" + attData.getValue()
-				        + "') for an attribute type that cannot be resolved by the following identifier: '"
-				        + attData.getKey() + "'");
+		// First, retrieve all attributes that are defined in the CSV and the values for the given row
+		Map<AT, Object> rowValues = new LinkedHashMap<>();
+		for (String header : line.getHeaderLine()) {
+			if (header.toLowerCase().startsWith(HEADER_ATTRIBUTE_PREFIX)) {
+				String attributeTypeIdentifier = StringUtils.removeStartIgnoreCase(header, HEADER_ATTRIBUTE_PREFIX);
+				AT attributeType = getAttributeType(attributeTypeIdentifier);
+				Object attributeValue = null;
+				String attributeValueStr = line.getString(header, "");
+				if (attributeType == null) {
+					throw new IllegalArgumentException("An attribute value is specified ('" + attributeValueStr
+					        + "') for an attribute type that cannot be resolved by the following identifier: '"
+					        + attributeType + "'");
```

Comment:
I guess this was meant to point to `attributeTypeIdentifier`? Would be nice having a unit test covering this - maybe not strictly necessary?

```suggestion
					        + attributeTypeIdentifier + "'");
```

Reply:
Good catch @Ruhanga . I've updated that.  This isn't new functionality, my main goal was to ensure all existing unit and integration tests pass, as well as any that I copied into the api-2.7 project, so my assumption is that the existing unit test coverage is sufficiently robust to handle a refactor.